Sources/PackageCollections/Utility.swift

```diff
@@ -0,0 +1,74 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import class Foundation.FileManager
+import TSCBasic
+
+import PackageModel
+import SourceControl
+
+// FIXME: Need a better place to put this + maybe better name
+struct Paths {
+    static func cache(fileSystem: FileSystem) -> AbsolutePath {
+        // use the idiomatic cache directory defined by FileManager when using the local file system
+        // otherwise use ~/.swift/cache
+        if fileSystem.isLocalFileSystem, let cache = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first {
+            return AbsolutePath(cache.path).appending(components: "org.swift.swiftpm")
+        }
+        return Self.dotSwiftPM(fileSystem: fileSystem).appending(component: "cache")
+    }
+
+    static func dotSwiftPM(fileSystem: FileSystem) -> AbsolutePath {
+        return fileSystem.homeDirectory.appending(component: ".swiftpm")
+    }
+}
+
+// FIXME: this is most likely wrong. is there a way to tell if an FS is the local FS / real FS?
+// it matters for things like SQLite or FileLock which expect to work on real FS
```

Comment:
👀

Reply:
Seems to me that FileLock should be implementable for any file system, but I agree that there needs to be some way to check this for things like SQLite.  Seems to me that this should be implemented in the FileSystem protocol itself, for the rare cases where it's needed.

Reply:
yea I feel its unfortunate that we need something like this, but there are case like SQLite that its needed. Do we want to move it to TSC then or keep here?

Reply:
Understood.  I think it's fine to keep it here.